plugins/audit-log-common/src/types.ts

```diff
@@ -0,0 +1,37 @@
+export type ActorDetails = {
+  actorId: string;
+  ip: string;
+  hostname: string;
+  userAgent: string;
+};
+
+// TODO: Get proper type for body
+export type AuditRequest = {
+  body: unknown;
+  uri: string;
+  method: string;
+};
+export type AuditResponse = {
+  body: unknown;
+  status: number;
+};
+export type auditLog = {
+  actor: ActorDetails;
+  eventName: string;
+  stage: string;
+  request?: AuditRequest;
+  response?: AuditResponse;
+  meta: { [key: string]: any };
+  isAuditLog: true;
+} & auditStatus;
```

Comment:
```suggestion
export type AuditResponse = {
  body: unknown;
  status: number;
};
export type auditLog = {
  actor: ActorDetails;
  eventName: string;
  stage: string;
  request?: Request;
  response?: Response;
  meta: { [key: string]: any };
  isAuditLog: true;
} & auditStatus;
```

Reply:
@schultzp2020 In the call I had with @AndrienkoAleksandr , we decided not to put the entire request/response object as to not expose tokens (from headers) and other irrelevant fields.

Reply:
For example, if I include the `Request` object in the audit log (note: the audit log structure is not the same as the one in this pull request), I get this extremely bloated audit log (I prettified it just to make it more readable):

```json
{
  "actor": {
    "client": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36",
    "ip_address": "::1",
    "user_id": "user:default/zaperex"
  },
  "event_name": "scaffolderTaskCreation",
  "isAuditLog": true,
  "level": "info",
  "message": "Scaffolding task for template:default/test-template with taskId: 3e7664d9-4400-4a66-825c-bde1ceaaf863 created by user:default/zaperex",
  "metadata": {
    "request": {
      "_body": true,
      "_consuming": false,
      "_dumped": false,
      "_events": {},
      "_eventsCount": 0,
      "_parsedUrl": {
        "_raw": "/v2/tasks",
        "auth": null,
        "hash": null,
        "host": null,
        "hostname": null,
        "href": "/v2/tasks",
        "path": "/v2/tasks",
        "pathname": "/v2/tasks",
        "port": null,
        "protocol": null,
        "query": null,
        "search": null,
        "slashes": null
      },
      "_readableState": {
        "autoDestroy": true,
        "awaitDrainWriters": null,
        "buffer": {
          "head": null,
          "length": 0,
          "tail": null
        },
        "closeEmitted": true,
        "closed": true,
        "constructed": true,
        "dataEmitted": true,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": true,
        "emitClose": true,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": true,
        "ended": true,
        "errorEmitted": false,
        "errored": null,
        "flowing": true,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": [],
        "readableListening": false,
        "reading": false,
        "readingMore": true,
        "resumeScheduled": false,
        "sync": true
      },
      "_remoteAddress": "::1",
      "_startAt": [
        51443,
        772574557
      ],
      "_startTime": "2024-05-08T18:54:10.461Z",
      "aborted": false,
      "baseUrl": "/api/scaffolder",
      "body": {
        "secrets": {
          "password": "sfgdhjkl"
        },
        "templateRef": "template:default/test-template",
        "values": {
          "password": "********",
          "username": "program",
          "wait_time": 10
        }
      },
      "client": {
        "_closeAfterHandlingError": false,
        "_events": {
          "close": [
            null,
            null,
            null,
            null
          ],
          "end": [
            null,
            null
          ],
          "error": [
            null,
            null
          ]
        },
        "_eventsCount": 8,
        "_hadError": false,
        "_host": null,
        "_httpMessage": {
          "_closed": false,
          "_contentLength": null,
          "_defaultKeepAlive": true,
          "_events": {
            "finish": [
              null,
              null,
              null
            ]
          },
          "_eventsCount": 2,
          "_expect_continue": false,
          "_hasBody": true,
          "_header": null,
          "_headerSent": false,
          "_keepAliveTimeout": 5000,
          "_last": false,
          "_maxRequestsPerSocket": 0,
          "_removedConnection": false,
          "_removedContLen": false,
          "_removedTE": false,
          "_sent100": false,
          "_trailer": "",
          "chunkedEncoding": false,
          "destroyed": false,
          "finished": false,
          "locals": {},
          "maxRequestsOnConnectionReached": false,
          "outputData": [],
          "outputSize": 0,
          "req": "[Circular]",
          "sendDate": true,
          "shouldKeepAlive": true,
          "socket": "[Circular]",
          "strictContentLength": false,
          "useChunkedEncodingByDefault": true,
          "writable": true
        },
        "_parent": null,
        "_paused": false,
        "_peername": {
          "address": "::1",
          "family": "IPv6",
          "port": 59434
        },
        "_pendingData": null,
        "_pendingEncoding": "",
        "_readableState": {
          "autoDestroy": true,
          "awaitDrainWriters": null,
          "buffer": {
            "head": null,
            "length": 0,
            "tail": null
          },
          "closeEmitted": false,
          "closed": false,
          "constructed": true,
          "dataEmitted": false,
          "decoder": null,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "emittedReadable": false,
          "encoding": null,
          "endEmitted": false,
          "ended": false,
          "errorEmitted": false,
          "errored": null,
          "flowing": true,
          "highWaterMark": 16384,
          "length": 0,
          "multiAwaitDrain": false,
          "needReadable": true,
          "objectMode": false,
          "pipes": [],
          "readableListening": false,
          "reading": true,
          "readingMore": false,
          "resumeScheduled": false,
          "sync": false
        },
        "_server": {
          "_connectionKey": "6::::7007",
          "_connections": 2,
          "_events": {
            "connection": [
              null,
              null
            ],
            "request": [
              null,
              null
            ]
          },
          "_eventsCount": 3,
          "_handle": {
            "reading": false
          },
          "_pendingSockets": {},
          "_unref": false,
          "_usingWorkers": false,
          "_workers": [],
          "allowHalfOpen": true,
          "connectionsCheckingInterval": 30000,
          "headersTimeout": 60000,
          "highWaterMark": 16384,
          "httpAllowHalfOpen": false,
          "keepAlive": false,
          "keepAliveInitialDelay": 0,
          "keepAliveTimeout": 5000,
          "maxHeadersCount": null,
          "maxRequestsPerSocket": 0,
          "noDelay": true,
          "pauseOnConnect": false,
          "rejectNonStandardBodyWrites": false,
          "requestTimeout": 300000,
          "timeout": 0
        },
        "_sockname": null,
        "_writableState": {
          "afterWriteTickInfo": null,
          "allBuffers": true,
          "allNoop": true,
          "autoDestroy": true,
          "bufferProcessing": false,
          "buffered": [],
          "bufferedIndex": 0,
          "closeEmitted": false,
          "closed": false,
          "constructed": true,
          "corked": 0,
          "decodeStrings": false,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "ended": false,
          "ending": false,
          "errorEmitted": false,
          "errored": null,
          "finalCalled": false,
          "finished": false,
          "highWaterMark": 16384,
          "length": 0,
          "needDrain": false,
          "objectMode": false,
          "pendingcb": 0,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false
        },
        "allowHalfOpen": true,
        "connecting": false,
        "parser": {
          "0": null,
          "_consumed": true,
          "_headers": [],
          "_url": "",
          "incoming": "[Circular]",
          "joinDuplicateHeaders": null,
          "maxHeaderPairs": 2000,
          "outgoing": null,
          "socket": "[Circular]"
        },
        "server": {
          "_connectionKey": "6::::7007",
          "_connections": 2,
          "_events": {
            "connection": [
              null,
              null
            ],
            "request": [
              null,
              null
            ]
          },
          "_eventsCount": 3,
          "_handle": {
            "reading": false
          },
          "_pendingSockets": {},
          "_unref": false,
          "_usingWorkers": false,
          "_workers": [],
          "allowHalfOpen": true,
          "connectionsCheckingInterval": 30000,
          "headersTimeout": 60000,
          "highWaterMark": 16384,
          "httpAllowHalfOpen": false,
          "keepAlive": false,
          "keepAliveInitialDelay": 0,
          "keepAliveTimeout": 5000,
          "maxHeadersCount": null,
          "maxRequestsPerSocket": 0,
          "noDelay": true,
          "pauseOnConnect": false,
          "rejectNonStandardBodyWrites": false,
          "requestTimeout": 300000,
          "timeout": 0
        },
        "timeout": 0
      },
      "complete": true,
      "httpVersion": "1.1",
      "httpVersionMajor": 1,
      "httpVersionMinor": 1,
      "joinDuplicateHeaders": null,
      "method": "POST",
      "originalUrl": "/api/scaffolder/v2/tasks",
      "params": {},
      "query": {},
      "rawHeaders": [
        "Host",
        "localhost:7007",
        "Connection",
        "keep-alive",
        "Content-Length",
        "151",
        "sec-ch-ua",
        "\"Chromium\";v=\"124\", \"Google Chrome\";v=\"124\", \"Not-A.Brand\";v=\"99\"",
        "Content-Type",
        "application/json",
        "sec-ch-ua-mobile",
        "?0",
        "authorization",
        "Bearer eyJ0eXAiOiJ2bmQuYmFja3N0YWdlLnVzZXIiLCJhbGciOiJFUzI1NiIsImtpZCI6ImY1N2Y2NGFlLTA0MzYtNDIyMC1hYWYxLWE0NDMyNDI0Mjg5MSJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjcwMDcvYXBpL2F1dGgiLCJzdWIiOiJ1c2VyOmRlZmF1bHQvemFwZXJleCIsImVudCI6WyJ1c2VyOmRlZmF1bHQvemFwZXJleCJdLCJhdWQiOiJiYWNrc3RhZ2UiLCJpYXQiOjE3MTUxOTQ0MzgsImV4cCI6MTcxNTE5ODAzOCwidWlwIjoiR3ZKWmo4bC0tdWVGTUhSV0dkVHNUY1dIVURmeEdWMEZEQmdtNEJRYkpvQllMOWdkSERvVWtmLWRWanhqLWI5VERvNndOWmMtcHItLWtqU1IxVV9BWkEifQ.sZ9ByvNsweIRi1z1HUmw1NljpiiauBvri0JL-f53yAB9UNCdscUpr0pZFhftPLhB7frfh0Gu1iTG35jZI28LPg",
        "User-Agent",
        "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36",
        "sec-ch-ua-platform",
        "\"Linux\"",
        "Accept",
        "*/*",
        "Origin",
        "http://localhost:3000",
        "Sec-Fetch-Site",
        "same-site",
        "Sec-Fetch-Mode",
        "cors",
        "Sec-Fetch-Dest",
        "empty",
        "Referer",
        "http://localhost:3000/",
        "Accept-Encoding",
        "gzip, deflate, br, zstd",
        "Accept-Language",
        "en-US,en;q=0.9"
      ],
      "rawTrailers": [],
      "res": {
        "_closed": false,
        "_contentLength": null,
        "_defaultKeepAlive": true,
        "_events": {
          "finish": [
            null,
            null,
            null
          ]
        },
        "_eventsCount": 2,
        "_expect_continue": false,
        "_hasBody": true,
        "_header": null,
        "_headerSent": false,
        "_keepAliveTimeout": 5000,
        "_last": false,
        "_maxRequestsPerSocket": 0,
        "_removedConnection": false,
        "_removedContLen": false,
        "_removedTE": false,
        "_sent100": false,
        "_trailer": "",
        "chunkedEncoding": false,
        "destroyed": false,
        "finished": false,
        "locals": {},
        "maxRequestsOnConnectionReached": false,
        "outputData": [],
        "outputSize": 0,
        "req": "[Circular]",
        "sendDate": true,
        "shouldKeepAlive": true,
        "socket": {
          "_closeAfterHandlingError": false,
          "_events": {
            "close": [
              null,
              null,
              null,
              null
            ],
            "end": [
              null,
              null
            ],
            "error": [
              null,
              null
            ]
          },
          "_eventsCount": 8,
          "_hadError": false,
          "_host": null,
          "_httpMessage": "[Circular]",
          "_parent": null,
          "_paused": false,
          "_peername": {
            "address": "::1",
            "family": "IPv6",
            "port": 59434
          },
          "_pendingData": null,
          "_pendingEncoding": "",
          "_readableState": {
            "autoDestroy": true,
            "awaitDrainWriters": null,
            "buffer": {
              "head": null,
              "length": 0,
              "tail": null
            },
            "closeEmitted": false,
            "closed": false,
            "constructed": true,
            "dataEmitted": false,
            "decoder": null,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": false,
            "emittedReadable": false,
            "encoding": null,
            "endEmitted": false,
            "ended": false,
            "errorEmitted": false,
            "errored": null,
            "flowing": true,
            "highWaterMark": 16384,
            "length": 0,
            "multiAwaitDrain": false,
            "needReadable": true,
            "objectMode": false,
            "pipes": [],
            "readableListening": false,
            "reading": true,
            "readingMore": false,
            "resumeScheduled": false,
            "sync": false
          },
          "_server": {
            "_connectionKey": "6::::7007",
            "_connections": 2,
            "_events": {
              "connection": [
                null,
                null
              ],
              "request": [
                null,
                null
              ]
            },
            "_eventsCount": 3,
            "_handle": {
              "reading": false
            },
            "_pendingSockets": {},
            "_unref": false,
            "_usingWorkers": false,
            "_workers": [],
            "allowHalfOpen": true,
            "connectionsCheckingInterval": 30000,
            "headersTimeout": 60000,
            "highWaterMark": 16384,
            "httpAllowHalfOpen": false,
            "keepAlive": false,
            "keepAliveInitialDelay": 0,
            "keepAliveTimeout": 5000,
            "maxHeadersCount": null,
            "maxRequestsPerSocket": 0,
            "noDelay": true,
            "pauseOnConnect": false,
            "rejectNonStandardBodyWrites": false,
            "requestTimeout": 300000,
            "timeout": 0
          },
          "_sockname": null,
          "_writableState": {
            "afterWriteTickInfo": null,
            "allBuffers": true,
            "allNoop": true,
            "autoDestroy": true,
            "bufferProcessing": false,
            "buffered": [],
            "bufferedIndex": 0,
            "closeEmitted": false,
            "closed": false,
            "constructed": true,
            "corked": 0,
            "decodeStrings": false,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": false,
            "ended": false,
            "ending": false,
            "errorEmitted": false,
            "errored": null,
            "finalCalled": false,
            "finished": false,
            "highWaterMark": 16384,
            "length": 0,
            "needDrain": false,
            "objectMode": false,
            "pendingcb": 0,
            "prefinished": false,
            "sync": false,
            "writecb": null,
            "writelen": 0,
            "writing": false
          },
          "allowHalfOpen": true,
          "connecting": false,
          "parser": {
            "0": null,
            "_consumed": true,
            "_headers": [],
            "_url": "",
            "incoming": "[Circular]",
            "joinDuplicateHeaders": null,
            "maxHeaderPairs": 2000,
            "outgoing": null,
            "socket": "[Circular]"
          },
          "server": {
            "_connectionKey": "6::::7007",
            "_connections": 2,
            "_events": {
              "connection": [
                null,
                null
              ],
              "request": [
                null,
                null
              ]
            },
            "_eventsCount": 3,
            "_handle": {
              "reading": false
            },
            "_pendingSockets": {},
            "_unref": false,
            "_usingWorkers": false,
            "_workers": [],
            "allowHalfOpen": true,
            "connectionsCheckingInterval": 30000,
            "headersTimeout": 60000,
            "highWaterMark": 16384,
            "httpAllowHalfOpen": false,
            "keepAlive": false,
            "keepAliveInitialDelay": 0,
            "keepAliveTimeout": 5000,
            "maxHeadersCount": null,
            "maxRequestsPerSocket": 0,
            "noDelay": true,
            "pauseOnConnect": false,
            "rejectNonStandardBodyWrites": false,
            "requestTimeout": 300000,
            "timeout": 0
          },
          "timeout": 0
        },
        "strictContentLength": false,
        "useChunkedEncodingByDefault": true,
        "writable": true
      },
      "route": {
        "methods": {
          "post": true
        },
        "path": "/v2/tasks",
        "stack": [
          {
            "keys": [],
            "method": "post",
            "name": "<anonymous>",
            "regexp": {
              "fast_slash": false,
              "fast_star": false
            }
          }
        ]
      },
      "socket": {
        "_closeAfterHandlingError": false,
        "_events": {
          "close": [
            null,
            null,
            null,
            null
          ],
          "end": [
            null,
            null
          ],
          "error": [
            null,
            null
          ]
        },
        "_eventsCount": 8,
        "_hadError": false,
        "_host": null,
        "_httpMessage": {
          "_closed": false,
          "_contentLength": null,
          "_defaultKeepAlive": true,
          "_events": {
            "finish": [
              null,
              null,
              null
            ]
          },
          "_eventsCount": 2,
          "_expect_continue": false,
          "_hasBody": true,
          "_header": null,
          "_headerSent": false,
          "_keepAliveTimeout": 5000,
          "_last": false,
          "_maxRequestsPerSocket": 0,
          "_removedConnection": false,
          "_removedContLen": false,
          "_removedTE": false,
          "_sent100": false,
          "_trailer": "",
          "chunkedEncoding": false,
          "destroyed": false,
          "finished": false,
          "locals": {},
          "maxRequestsOnConnectionReached": false,
          "outputData": [],
          "outputSize": 0,
          "req": "[Circular]",
          "sendDate": true,
          "shouldKeepAlive": true,
          "socket": "[Circular]",
          "strictContentLength": false,
          "useChunkedEncodingByDefault": true,
          "writable": true
        },
        "_parent": null,
        "_paused": false,
        "_peername": {
          "address": "::1",
          "family": "IPv6",
          "port": 59434
        },
        "_pendingData": null,
        "_pendingEncoding": "",
        "_readableState": {
          "autoDestroy": true,
          "awaitDrainWriters": null,
          "buffer": {
            "head": null,
            "length": 0,
            "tail": null
          },
          "closeEmitted": false,
          "closed": false,
          "constructed": true,
          "dataEmitted": false,
          "decoder": null,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "emittedReadable": false,
          "encoding": null,
          "endEmitted": false,
          "ended": false,
          "errorEmitted": false,
          "errored": null,
          "flowing": true,
          "highWaterMark": 16384,
          "length": 0,
          "multiAwaitDrain": false,
          "needReadable": true,
          "objectMode": false,
          "pipes": [],
          "readableListening": false,
          "reading": true,
          "readingMore": false,
          "resumeScheduled": false,
          "sync": false
        },
        "_server": {
          "_connectionKey": "6::::7007",
          "_connections": 2,
          "_events": {
            "connection": [
              null,
              null
            ],
            "request": [
              null,
              null
            ]
          },
          "_eventsCount": 3,
          "_handle": {
            "reading": false
          },
          "_pendingSockets": {},
          "_unref": false,
          "_usingWorkers": false,
          "_workers": [],
          "allowHalfOpen": true,
          "connectionsCheckingInterval": 30000,
          "headersTimeout": 60000,
          "highWaterMark": 16384,
          "httpAllowHalfOpen": false,
          "keepAlive": false,
          "keepAliveInitialDelay": 0,
          "keepAliveTimeout": 5000,
          "maxHeadersCount": null,
          "maxRequestsPerSocket": 0,
          "noDelay": true,
          "pauseOnConnect": false,
          "rejectNonStandardBodyWrites": false,
          "requestTimeout": 300000,
          "timeout": 0
        },
        "_sockname": null,
        "_writableState": {
          "afterWriteTickInfo": null,
          "allBuffers": true,
          "allNoop": true,
          "autoDestroy": true,
          "bufferProcessing": false,
          "buffered": [],
          "bufferedIndex": 0,
          "closeEmitted": false,
          "closed": false,
          "constructed": true,
          "corked": 0,
          "decodeStrings": false,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "ended": false,
          "ending": false,
          "errorEmitted": false,
          "errored": null,
          "finalCalled": false,
          "finished": false,
          "highWaterMark": 16384,
          "length": 0,
          "needDrain": false,
          "objectMode": false,
          "pendingcb": 0,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false
        },
        "allowHalfOpen": true,
        "connecting": false,
        "parser": {
          "0": null,
          "_consumed": true,
          "_headers": [],
          "_url": "",
          "incoming": "[Circular]",
          "joinDuplicateHeaders": null,
          "maxHeaderPairs": 2000,
          "outgoing": null,
          "socket": "[Circular]"
        },
        "server": {
          "_connectionKey": "6::::7007",
          "_connections": 2,
          "_events": {
            "connection": [
              null,
              null
            ],
            "request": [
              null,
              null
            ]
          },
          "_eventsCount": 3,
          "_handle": {
            "reading": false
          },
          "_pendingSockets": {},
          "_unref": false,
          "_usingWorkers": false,
          "_workers": [],
          "allowHalfOpen": true,
          "connectionsCheckingInterval": 30000,
          "headersTimeout": 60000,
          "highWaterMark": 16384,
          "httpAllowHalfOpen": false,
          "keepAlive": false,
          "keepAliveInitialDelay": 0,
          "keepAliveTimeout": 5000,
          "maxHeadersCount": null,
          "maxRequestsPerSocket": 0,
          "noDelay": true,
          "pauseOnConnect": false,
          "rejectNonStandardBodyWrites": false,
          "requestTimeout": 300000,
          "timeout": 0
        },
        "timeout": 0
      },
      "statusCode": null,
      "statusMessage": null,
      "upgrade": false,
      "url": "/v2/tasks"
    },
    "taskId": "3e7664d9-4400-4a66-825c-bde1ceaaf863",
    "templateRef": "template:default/test-template"
  },
  "plugin": "scaffolder",
  "service": "backstage",
  "status": "success",
  "timestamp": "2024-05-08T18:54:10.480Z",
  "type": "plugin"
}
```

Reply:
Hmm, just noticed that the secrets inputted into the scaffolder software template was set in clear text in the body object. Need to figure out a way to redact it or maybe just not include the body at all?

Reply:
@schultzp2020, Apologies, but the Kubernetes (k8s) audit log doesn't track all requests and responses – they are indeed quite large. The k8s audit log only records request/response bodies. The audit log file will grow very quickly if we include the entire request/response. I anticipate that the audit log streamer will fail to write data when there is no more free disk space available.

Reply:
@Zaperex , I agree with you request/response body could contain sensitive information. I think that's why k8s audit log has level feature. Level values:

```
None - don't log events that match this rule.
Metadata - log request metadata (requesting user, timestamp, resource, verb, etc.) but not request or response body.
Request - log event metadata and request body but not response body. This does not apply for non-resource requests.
RequestResponse - log event metadata, request and response bodies. This does not apply for non-resource requests.
```

@schultzp2020 , @Zaperex Since we don't have strong agreement about request/response body, we can postpone this audit log feature and apply it in the next iteration. We need to consider how to hide sensitive information and we have a lack time for RBAC audit log. And maybe it make sense to have request/response body audit log message like optional feature. We can introduce auditLog application configuration and provide the way how enable/disable request/response body audit log messages. Each plugin with audit log implementation can use this configuration.



@Zaperex, I agree with your assessment that the request/response body could contain sensitive information. I believe this is why the k8s audit log incorporates a level feature. The level values are as follows:

```
None - don't log events that match this rule.
Metadata - log request metadata (requesting user, timestamp, resource, verb, etc.) but not the request or response body.
Request - log event metadata and the request body but not the response body. This does not apply to non-resource requests.
RequestResponse - log event metadata, the request, and response bodies. This does not apply to non-resource requests.
```

Admins can configure the system to track only Metadata audit log lines, without including the request/response.

@schultzp2020 , @Zaperex Since given the lack of strong consensus regarding the handling of request/response bodies, I suggest postponing this audit log feature and implementing it in the next iteration. We need to carefully consider how to hide sensitive information, and currently, our priority lies with RBAC audit logs due to time constraints. Additionally, it might be sensible to make the request/response body audit log message an optional feature. We can introduce an auditLog application configuration and provide a way to enable/disable request/response body audit log messages. Each plugin with audit log implementation can utilize this configuration.

So admin can configure to track only Metadata audit log lines, without request/response audit log lines.

Reply:
For now, I just added a comment for users to redact possibly sensitive information in the `req.body` before sending it to be audit logged.